bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterBridgeHandler.java

```diff
@@ -0,0 +1,169 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.meater.internal.MeaterBridgeConfiguration;
+import org.openhab.binding.meater.internal.api.MeaterRestAPI;
+import org.openhab.binding.meater.internal.discovery.MeaterDiscoveryService;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link MeaterBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeaterBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private int refreshTimeInSeconds = 300;
+
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private final TranslationProvider i18nProvider;
+    private LocaleProvider localeProvider;
+    private final Map<String, MeaterProbeDTO.Device> meaterProbeThings = new ConcurrentHashMap<>();
+
+    private @Nullable MeaterRestAPI api;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public MeaterBridgeHandler(Bridge bridge, HttpClient httpClient, Gson gson, TranslationProvider i18nProvider,
+            LocaleProvider localeProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = gson;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+    }
+
+    @Override
+    public void initialize() {
+        MeaterBridgeConfiguration config = getConfigAs(MeaterBridgeConfiguration.class);
+
+        api = new MeaterRestAPI(config, gson, httpClient, localeProvider);
+        refreshTimeInSeconds = config.refresh;
+
+        if (config.email == null || config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/config.missing-username-password.description");
+        } else if (refreshTimeInSeconds < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/config.negative-refresh-interval.description");
```

Comment:
Normally, this check is now no more required because the core framework now checks the validity of your thing configuration parameters before starting the handler.

Reply:
You are correct that the check on refresh time is unnecessary, I have removed it and also the i18n representation.

The test on null values on email/password are also unecessary since that is not possible to configure, I also removed the @Nullable fields on username/password and initialized them to empty string.

However if you configure empty string that will lead to  an unnecessary  login attempt . 
I therefore changed the null checks to empty string checks with same warning.